src/app/Plans/components/Wizard/helpers.tsx

```diff
@@ -351,32 +355,45 @@ export const filterSourcesBySelectedVMs = (
   availableSources: MappingSource[],
   selectedVMs: SourceVM[],
   mappingType: MappingType,
-  sourceProviderType: ProviderType
+  sourceProviderType: ProviderType,
+  nicProfiles: INicProfile[],
+  disks: IDisk[]
 ): MappingSource[] => {
-  const sourceIds = Array.from(
+  const sourceIds: (string | undefined)[] = Array.from(
     new Set(
       selectedVMs.flatMap((vm) => {
         if (mappingType === MappingType.Network) {
           if (sourceProviderType === 'vsphere') {
             return (vm as IVMwareVM).networks.map((network) => network.id);
           }
           if (sourceProviderType === 'ovirt') {
-            return (vm as IRHVVM).nics.map((nic) => nic.profile.network);
+            const networkIds = (vm as IRHVVM).nics.map((nic) => {
```

Comment:
Nitpick, but if you want to be consistent with storages in terms of doing this in 2 steps for readability, you could do:

```ts
const vmNicProfiles = (vm as IRHVVM).nics.map(nic) =>
  nicProfiles.find((nicProfile) => nicProfile.id === nic.profile)
);
const networkIds = vmNicProfiles.map((nicProfile => nicProfile?.network);
return networkIds;
```